From 87715cc4917e0edcc1090658ea09be508f47f3b7 Mon Sep 17 00:00:00 2001
From: Timo Jyrinki <timo.jyrinki@canonical.com>
Date: Wed, 23 Mar 2016 09:41:50 +0000
Subject: [PATCH] Blacklist Mali-T760/Mali-T720 from supporting BGRA.

Extend the blacklist with Mali-T760 and Mali-T720 as found on Meizu
Pro 5 and Bq Aquaris M10 Ubuntu Editions. Reading from FBO like
taking screenshots does not produce correct result otherwise.

Initially reported at: https://launchpad.net/bugs/1557915 and
https://launchpad.net/bugs/1559906

Change-Id: Ic875bd083277bf933863a3a50f8e874dd6e04365
Reviewed-by: Laszlo Agocs <laszlo.agocs@theqtcompany.com>
---
 src/gui/opengl/qopenglframebufferobject.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/gui/opengl/qopenglframebufferobject.cpp b/src/gui/opengl/qopenglframebufferobject.cpp
index 5184283..0e1074f 100644
--- a/src/gui/opengl/qopenglframebufferobject.cpp
+++ b/src/gui/opengl/qopenglframebufferobject.cpp
@@ -1281,9 +1281,13 @@ static inline QImage qt_gl_read_framebuffer_rgba8(const QSize &size, bool includ
     const char *renderer = reinterpret_cast<const char *>(funcs->glGetString(GL_RENDERER));
     const char *ver = reinterpret_cast<const char *>(funcs->glGetString(GL_VERSION));
 
-    // Blacklist PowerVR Rogue G6200 as it has problems with its BGRA support.
+    // Blacklist GPU chipsets that have problems with their BGRA support.
     const bool blackListed = (qstrcmp(renderer, "PowerVR Rogue G6200") == 0
-                             && ::strstr(ver, "1.3") != 0);
+                             && ::strstr(ver, "1.3") != 0) ||
+                             (qstrcmp(renderer, "Mali-T760") == 0
+                             && ::strstr(ver, "3.1") != 0) ||
+                             (qstrcmp(renderer, "Mali-T720") == 0
+                             && ::strstr(ver, "3.1") != 0);
 
     const bool supports_bgra = has_bgra_ext && !blackListed;
 
-- 
2.7.4

